code/BuddyTrackerBasic/src/main.cpp

```diff
@@ -99,25 +102,28 @@ void sendPacket(BT_Packet packet){
 // currently adds unknown buddies
 // shows everyone
 void updateBuddy(uint64_t UUID, uint16_t lat_partial, uint16_t lng_partial){
-    uint8_t index = findBuddyIndex(UUID);
+    uint8_t index = findBuddyBy(UUID);
     
     // add unknown buddies
     if(index == -1){
-        Buddy *newBuddy = new Buddy(UUID);
-        buddies.add(newBuddy);
-        index = buddies.size() - 1;
+        if (buddies.size() < MAX_UINT8) {
```

Comment:
Don't assume that the list is managed elsewhere. Best to check size before adding.